pkg/ocicni/types.go

```diff
@@ -75,22 +75,33 @@ type PodNetwork struct {
 	// NetNS is the network namespace path of the sandbox.
 	NetNS string
 
-	// Networks is a list of CNI network names to attach to the sandbox
-	// Leave this list empty to attach the default network to the sandbox
-	Networks []string
+	// Networks is a list of CNI network names (and optional interface
+	// names) to attach to the sandbox. Leave this list empty to attach the
+	// default network to the sandbox
+	Networks []NetAttachment
 
 	// NetworkConfig is configuration specific to a single CNI network.
 	// It is optional, and can be omitted for some or all specified networks
 	// without issue.
 	RuntimeConfig map[string]RuntimeConfig
 }
 
+// NetAttachment describes a container network attachment
+type NetAttachment struct {
+	// NetName contains the name of the CNI network to which the container
+	// should be or is attached
+	Name string
+	// Ifname contains the optional interface name of the attachment
+	Ifname string
+}
+
 // NetResult contains the result the network attachment operation
 type NetResult struct {
 	// Result is the CNI Result
 	Result types.Result
-	// Ifname is the container interface name for this network attachment
-	Ifname string
+	// AttachInfo contains the network and interface names of this
+	// network attachment
+	AttachInfo NetAttachment
```

Comment:
How about just extending **Network Name** filed in NetResult? Seems more compatible.

Reply:
@dcbw   I mean just extend NetResult like 
```
type NetResult struct {
	Result types.Result
	NetAttachment
}
```
Then old users can still use NetResult.Ifname instead of NetResult.AttachInfo.Ifname. Seems more break-less.

Reply:
@mars1024 ah yes, good catch